### PR TITLE
Add url unfurling for links (Youtube only for now)

### LINKED ISSUE #1235: URL unfurling
https://github.com/status-im/status/issues/2

see relevant equivalent PR https://github.com/status-im/status-react/pull/11311/files

### PR DESCRIPTION
**Pull request by jrainville**
Fixes #1235

Only supports Youtube for now, because that's what Status-go supports, but it will support more at some point

I put it as Draft because firstly, it's super ugly. @simonricoo probably has something pretty cool cooking for it.
Secondly, the status-go commit is a branch/PR and not develop, so we might lose the branch once the PR is merged.

image
 (You'll note that the youtu.be link didn't unfurl. That's because status-go doesn't support it yet)

image

### CODE REVIEW

File: src/status/libstatus/settings.nim [hunk 1]
```diff
@@ -65,4 +65,7 @@ proc getCurrentNetwork*(): Network =
 proc getCurrentNetworkDetails*(): NetworkDetails =
   let currNetwork = getSetting[string](Setting.Networks_CurrentNetwork, constants.DEFAULT_NETWORK_NAME)
   let networks = getSetting[seq[NetworkDetails]](Setting.Networks_Networks)
-  networks.find((network: NetworkDetails) => network.id == currNetwork)
\ No newline at end of file
+  networks.find((network: NetworkDetails) => network.id == currNetwork)
+    
+proc getLinkPreviewWhitelist*(): JsonNode =
+  result = callPrivateRPC("getLinkPreviewWhitelist".prefix, %* []).parseJSON()["result"]
```
Comment (richard-ramos) [inline src/status/libstatus/settings.nim:71]: Is it possible that `getLinkPreviewWhitelist` fails?

**Review by richard-ramos — APPROVED** [review]
Very cool!

**Comment by simonricoo** [comment]
@jrainville Here is the url unfurling design: https://www.figma.com/file/Mr3rqxxgKJ2zMQ06UAKiWL/%F0%9F%92%AC-Chat%E2%8E%9CDesktop?node-id=3038%3A7769

File: src/status/libstatus/settings.nim (hunk 1 above)
Comment (jrainville) [reply src/status/libstatus/settings.nim:71]: Ah, probably. I'll add a try/catch when I work on the design

**Comment by jrainville** [comment]
I added a new commit with the design implemented (most of it at least)

**Pull request merged**